File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi=float('-inf')
        add=0
        for ele in nums:
            add+=ele
            if add>maxi:
                maxi=add
            if add<0:
                add=0
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max=Integer.MIN_VALUE;
        int n=nums.length;
        int sum=0;
        for(int i=0;i<n;i++){
            sum+=nums[i];
            if(max<sum) max=sum;
            if(sum<0) sum=0;
        }
        return max;
    }
}
